### DISCUSSION TITLE
Use whisper to automatically generate subtitles and support translation!

### DISCUSSION THREAD

**Discussion by qinL-cdy** [openai/whisper#1659] [Show and tell]
go see https://github.com/qinL-cdy/auto_ai_subtitle for details

支持音频提取，文字提取，字幕生成，字幕翻译等功能
Support audio extraction, text extraction, subtitle generation, subtitle translation and other functions

感谢社区的贡献，现在可以支持hugging face上下载的自定义模型
Thanks to the community's contributions, custom models downloaded on Hugging Face are supported

**Comment by Pareek-Yash** [comment]
Hi there! I am working on something similar but using MarianMT for this task, I have tried various ways of using MarianMT, can you guide me somehow? I have tried translating text one line after another, separated by timestamps, which leads to loss of context sometimes, and tried connecting text separated by [SEP] or <span> tags, which works for some languages and busts for others. I have no experience in MT so would be great help.
